### ISSUE TITLE
Mopidy-TuneIn on cyrillic input in search

### ISSUE THREAD

**Issue by sheppardv** [pimusicbox/pimusicbox#190]
```
{
   "jsonrpc":"2.0",
   "id":13,
   "error":{
      "message":"Application error",
      "code":0,
      "data":{
         "message":"u'\\u0415'",
         "traceback":"Traceback (most recent call last):\n  File \"/usr/local/lib/python2.7/dist-packages/mopidy/utils/jsonrpc.py\", line 130, in _handle_single_request\n    result = self._unwrap_result(result)\n  File \"/usr/local/lib/python2.7/dist-packages/mopidy/utils/jsonrpc.py\", line 219, in _unwrap_result\n    result = result.get()\n  File \"/usr/lib/python2.7/dist-packages/pykka/future.py\", line 299, in get\n    exec('raise exc_info[0], exc_info[1], exc_info[2]')\n  File \"/usr/lib/python2.7/dist-packages/pykka/actor.py\", line 200, in _actor_loop\n    response = self._handle_receive(message)\n  File \"/usr/lib/python2.7/dist-packages/pykka/actor.py\", line 294, in _handle_receive\n    return callee(*message['args'], **message['kwargs'])\n  File \"/usr/local/lib/python2.7/dist-packages/mopidy/core/library.py\", line 195, in search\n    return [result for result in pykka.get_all(futures) if result]\n  File \"/usr/lib/python2.7/dist-packages/pykka/future.py\", line 330, in get_all\n    return [future.get(timeout=timeout) for future in futures]\n  File \"/usr/lib/python2.7/dist-packages/pykka/future.py\", line 299, in get\n    exec('raise exc_info[0], exc_info[1], exc_info[2]')\n  File \"/usr/lib/python2.7/dist-packages/pykka/actor.py\", line 200, in _actor_loop\n    response = self._handle_receive(message)\n  File \"/usr/lib/python2.7/dist-packages/pykka/actor.py\", line 294, in _handle_receive\n    return callee(*message['args'], **message['kwargs'])\n  File \"/usr/local/lib/python2.7/dist-packages/mopidy_tunein/actor.py\", line 96, in search\n    tunein_query = translator.mopidy_to_tunein_query(query)\n  File \"/usr/local/lib/python2.7/dist-packages/mopidy_tunein/translator.py\", line 91, in mopidy_to_tunein_query\n    return urllib.pathname2url(' '.join(tunein_query))\n  File \"/usr/lib/python2.7/urllib.py\", line 60, in pathname2url\n    return quote(pathname)\n  File \"/usr/lib/python2.7/urllib.py\", line 1268, in quote\n    return ''.join(map(quoter, s))\nKeyError: u'\\u0415'\n",
         "type":"KeyError"
      }
   }
}
```

**Comment by jodal** [comment]
The traceback cleaned up:

``` python
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/mopidy/utils/jsonrpc.py", line 130, in _handle_single_request
    result = self._unwrap_result(result)
  File "/usr/local/lib/python2.7/dist-packages/mopidy/utils/jsonrpc.py", line 219, in _unwrap_result
    result = result.get()
  File "/usr/lib/python2.7/dist-packages/pykka/future.py", line 299, in get
    exec('raise exc_info[0], exc_info[1], exc_info[2]')
  File "/usr/lib/python2.7/dist-packages/pykka/actor.py", line 200, in _actor_loop
    response = self._handle_receive(message)
  File "/usr/lib/python2.7/dist-packages/pykka/actor.py", line 294, in _handle_receive
    return callee(*message['args'], **message['kwargs'])
  File "/usr/local/lib/python2.7/dist-packages/mopidy/core/library.py", line 195, in search
    return [result for result in pykka.get_all(futures) if result]
  File "/usr/lib/python2.7/dist-packages/pykka/future.py", line 330, in get_all
    return [future.get(timeout=timeout) for future in futures]
  File "/usr/lib/python2.7/dist-packages/pykka/future.py", line 299, in get
    exec('raise exc_info[0], exc_info[1], exc_info[2]')
  File "/usr/lib/python2.7/dist-packages/pykka/actor.py", line 200, in _actor_loop
    response = self._handle_receive(message)
  File "/usr/lib/python2.7/dist-packages/pykka/actor.py", line 294, in _handle_receive
    return callee(*message['args'], **message['kwargs'])
  File "/usr/local/lib/python2.7/dist-packages/mopidy_tunein/actor.py", line 96, in search
    tunein_query = translator.mopidy_to_tunein_query(query)
  File "/usr/local/lib/python2.7/dist-packages/mopidy_tunein/translator.py", line 91, in mopidy_to_tunein_query
    return urllib.pathname2url(' '.join(tunein_query))
  File "/usr/lib/python2.7/urllib.py", line 60, in pathname2url
    return quote(pathname)
  File "/usr/lib/python2.7/urllib.py", line 1268, in quote
    return ''.join(map(quoter, s))
KeyError: u'\u0415'
```

**Comment by jodal** [comment]
@kingosticks Does Mopidy-TuneIn make sure to encode the `tunein_query` from Unicode to UTF-8-encoded bytes before passing it to `urllib.pathname2url()`?

**Comment by kingosticks** [comment]
Sure looks that way: https://github.com/kingosticks/mopidy-tunein/blob/develop/mopidy_tunein/translator.py#L102

This is quite an old report, I'm not convinced it exists in the latest version.

**Comment by jodal** [comment]
That code looks good. Seems to be fixed in kingosticks/mopidy-tunein@35c7617bd9ff0916b1c0203a034768d9e9d6eb4a, which is included since v0.2.0, released 2015-03-26. That's 10 days before Pi MusicBox 0.6.0 was released, so I'm assuming that this is fixed in 0.6.0. Closing.
